src/main/java/io/odpf/firehose/sink/mongodb/request/MongoRequestHandler.java

```diff
@@ -0,0 +1,53 @@
+package io.odpf.firehose.sink.mongodb.request;
+
+
+import com.mongodb.client.model.WriteModel;
+import io.odpf.firehose.config.enums.MongoSinkMessageType;
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.exception.JsonParseException;
+import io.odpf.firehose.serializer.MessageToJson;
+import org.bson.Document;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.nio.charset.Charset;
+
+public abstract class MongoRequestHandler {
+    private final MongoSinkMessageType messageType;
+    private final MessageToJson jsonSerializer;
+    private final JSONParser jsonParser;
+
+    public MongoRequestHandler(MongoSinkMessageType messageType, MessageToJson jsonSerializer) {
+        this.messageType = messageType;
+        this.jsonSerializer = jsonSerializer;
+        this.jsonParser = new JSONParser();
+    }
+
+    public abstract boolean canCreate();
+
+    public abstract WriteModel<Document> getRequest(Message message);
+
+    String extractPayload(Message message) {
```

Comment:
Put access modifiers.. should be 'protected'.

Reply:
getRequest() is invoked in MongoSink class in package io.odpf.firehose.sink.mongodb
MongoRequestHandler is in package io.odpf.firehose.sink.mongodb.request
thus need public access to access outside package

Reply:
sure you can put public.